File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.js
// cards1 또는 cards2의 0번째 index를 goal의 0번째 index와 비교해서 같다면 제거 
// goal의 length가 0이 될 경우 Yes를 return
// cards1 또는 cards2의 0번째 index를 goal의 0번째 index와 비교해서 둘 다 같지 않다면 No를 return

function solution(cards1, cards2, goal) {
  
  while (goal.length > 0) {
    if (cards1[0] === goal[0]) {
      cards1.shift()
      goal.shift()
    } else if (cards2[0] === goal[0]) {
      cards2.shift()
      goal.shift()
    } else return "No"
  }
  return "Yes"
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):

    # 아래처럼 0 안 넣어주면, 아래 사례에서 cards2가 길이가 0이되고,
        # elif문에서 cards2[0] -> list index out of range에러가 발생
    cards1.append(0)
    cards2.append(0)
    
    for k in range(len(goal)):
        if goal[k] == cards1[0]:
            cards1.pop(0)
        elif goal[k] == cards2[0]:
            cards2.pop(0)
        else:
            return 'No'
    
    return 'Yes'


print(solution(["i", "water", "drink"],["want", "to"],["i", "want", "to", "drink", "water"]))